nucypher/blockchain/eth/networks.py

```diff
@@ -1,57 +1,68 @@
-        LYNX: 5,  # Goerli
-        TAPIR: 5,
-        ORYX: 5
-    }
-    __to_chain_id_polygon = {
-        # TODO: Use naming scheme?
-        POLYGON: 137,    # Polygon Mainnet
-        MUMBAI: 80001,   # Polygon Testnet (Mumbai)
-    }
-
-    ETH_NETWORKS = tuple(__to_chain_id_eth.keys())
-    POLY_NETWORKS = tuple(__to_chain_id_polygon.keys())
-
-    NETWORKS = ETH_NETWORKS + POLY_NETWORKS
-
-    class UnrecognizedNetwork(RuntimeError):
-        pass
+from enum import Enum
+from typing import NamedTuple
 
-    @classmethod
-    def get_ethereum_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_eth[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+from nucypher.config.constants import TEMPORARY_DOMAIN
 
-    @classmethod
-    def get_polygon_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_polygon[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+
+class ChainInfo(NamedTuple):
+    id: int
+    name: str
+
+
+class EthChain(ChainInfo, Enum):
+    MAINNET = ChainInfo(1, "mainnet")
+    GOERLI = ChainInfo(5, "goerli")
+    SEPOLIA = ChainInfo(11155111, "sepolia")
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class PolygonChain(ChainInfo, Enum):
+    POLYGON = ChainInfo(137, "polygon")
+    MUMBAI = ChainInfo(80001, "mumbai")
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class TACoDomain(NamedTuple):
+    name: str
+    eth_chain: EthChain
+    polygon_chain: PolygonChain
+
+    def is_testnet(self) -> bool:
+        return self.eth_chain != EthChain.MAINNET
+
+
+class UnrecognizedDomain(RuntimeError):
+    """Raised when a provided domain name is not recognized."""
+
+
+class NetworksInventory:
+    MAINNET = TACoDomain("mainnet", EthChain.MAINNET, PolygonChain.POLYGON)
+    # Testnets
+    ORYX = TACoDomain("oryx", EthChain.GOERLI, PolygonChain.POLYGON)
+    LYNX = TACoDomain("lynx", EthChain.GOERLI, PolygonChain.MUMBAI)
+    TAPIR = TACoDomain("tapir", EthChain.SEPOLIA, PolygonChain.MUMBAI)
+    IBEX = TACoDomain(
+        "ibex", EthChain.GOERLI, None
+    )  # this is required for configuration file migrations (backwards compatibility)
+
+    SUPPORTED_DOMAINS = [
+        MAINNET,
+        ORYX,
+        LYNX,
+        TAPIR,
+    ]
+
+    SUPPORTED_DOMAIN_NAMES = [network.name for network in SUPPORTED_DOMAINS]
+
+    # TODO not needed once merged with registry changes
+    POLYGON_CHAINS = [network.polygon_chain.name for network in SUPPORTED_DOMAINS]
+
+    DEFAULT_DOMAIN_NAME: str = MAINNET.name
 
     @classmethod
-    def validate_network_name(cls, network_name: str):
-        if network_name not in cls.NETWORKS:
-            raise cls.UnrecognizedNetwork(
-                f"{network_name} is not a recognized network."
-            )
+    def from_domain_name(cls, domain: str) -> TACoDomain:
```

Comment:
This method name implies that it's an alternate constructor, but it's not (it's a resolver).  I think this can this be simplified by removing `NetworksInventory` all together - There are too many namespaces and levels of indirection here IMO.  I'd like to encourage simple module-level constants instead.

Reply:
Oh. I think I misunderstood what you originally meant. Yes, this change is indeed being made - it's on my list. Changes incoming.

Reply:
✔️